docs/whitepaper/src/appendix.tex

```diff
@@ -0,0 +1,237 @@
+    \end{algorithmic}\label{alg:algorithm-1}
+\end{algorithm}
+
+
+\begin{algorithm}[ht]
+    \begin{algorithmic}[1]
+        \caption{$\textsf{PrepareForBlockValidation}(sl_j, v_i^{\text{vrf}}, C_{\text{loc}})$}
+        \State \lstinline|// Parse and calculate all necessary values for block validation.|
+        \State Calculate $e_n$ for the given $sl_j$
+        \State \lstinline|// Main ledger state_m is calculated according to the last block|
+        \State \lstinline|// produced up to m-th slot.|
+        \State Parse $\textsf{state}_{(n - 4)\cdot R},
+        \textsf{state}_{(n - 2)\cdot R}$ from $C_{\text{loc}}$
+        \State Parse verified participants table $V{\text{ver}}$ from $\textsf{state}_{(n - 4)\cdot R}$
+        \State Parse an associated committee id $k$ for the given $v_i^{\text{vrf}}$ from $V{\text{ver}}$
+        \State \lstinline|// Set epoch randomness:|
+        \State Set ${\eta_{n-2} \leftarrow \mathcal{F}_{\text{LB}}(e_{n-2}, C_{\text{loc}})}$ \lstinline|// For the consensus lottery.|
+        \State Set ${\eta_{n} \leftarrow \mathcal{F}_{\text{LB}}(e_{n}, C_{\text{loc}})}$ \lstinline|// For the leader lottery.|
+        \State \lstinline|// Get stakeholders distribution used for the leader lottery:|
+        \State Parse $k$-th committee stakeholders distribution $S_k^{\text{cons}, n - 2}$ from the $\textsf{state}_{(n - 2)\cdot R}$
+        \State \lstinline|// Get stakeholders distribution used for the consensus group lottery:|
+        \State Parse verified and equipped with $\mathcal{F}^k_{\text{ConnSys}}$ functionality stakeholders distribution $S_k^{\text{ver}, {n - 4}}$
+        from $\textsf{state}_{(n - 4)\cdot R}$
+        \State \lstinline|// Set lotteries thresholds:|
+        \State Set participant's consensus group lottery threshold for $k$-th committee as $T_{i, n-2}^{\text{cons}, k}$ calculated using $S_k^{\text{ver}, {n - 4}}$
+        \State Set participant's leader lottery threshold as $T_{i, n}^{\text{lead}, k}$ calculated using $S_k^{\text{cons}, {n - 2}}$
+        \State Set participant's synchronization lottery threshold as $T_{i, n}^{\text{sync}}$ calculated using $S_k^{\text{cons}, {n - 2}}$
+
+        \Return ${(e_n, V_{\text{ver}}, \eta_{n}, \eta_{n - 2}, T_{i, n-2}^{\text{cons}, k}, T_{i, n}^{\text{lead}, k}, T_{i, n}^{\text{sync}})}$
+    \end{algorithmic}\label{alg:algorithm-2.0}
+\end{algorithm}
+
+\begin{algorithm}[ht]
+    \begin{algorithmic}[1]
+        \caption{$\textsf{IsValidBlock}(B, C_{text{\loc}})$}
+        \State \lstinline|// All indexes except the epoch index are omitted.|
+        \State Parse $B$ as ${(h, sl, \textsf{state}, v^{\text{vrf}}, \pi^{\text{sl}}, \sigma_{\text{KES}})}$ \lstinline|// value h is a block body hash.|
+        \State \lstinline|// Prepare constants:|
+
+        \State $(e_n, V_{\text{ver}}, \eta_{n}, \eta_{n - 2}, T^{\text{cons}}, T^{\text{lead}, T^{\text{sync}}) = \textsf{PrepareForBlockValidation}(sl, v^{\text{vrf}}, C_{\text{loc}})$
```

Comment:
Looks like eta_{n-2} isn't used down the road

Reply:
One check was missed, fixed